src/libmongoc/src/mongoc/mongoc-client-side-encryption.c

```diff
@@ -50,6 +50,7 @@ struct _mongoc_auto_encryption_opts_t {
    bool bypass_query_analysis;
    mc_kms_credentials_callback creds_cb;
    bson_t *extra;
+   int64_t cache_expiration_ms; /* <0 means unset */
```

Comment:
```suggestion
   mcd_optional_u64_t cache_expiration_ms;
```

For consistency with the libmongocrypt's `mongocrypt_setopt_key_expiration`, suggest accepting and storing as a `uint64_t` to avoid integer conversions. Suggest defining a new optional type in `mongoc-util-private.h`:

```c
// `mcd_optional_u64_t` is an optional `uint64_t`.
typedef struct {
   bool isset;
   uint64_t value;
} mcd_optional_u64_t;
```




And assign with a compound literal:
```c
opts->cache_expiration_ms = (mcd_optional_u64_t){.isset = true, .value = expiration};
```